### PR TITLE
WIP: add shoutrrr support

### PR DESCRIPTION
**Pull request by mbrandau**
I copy and pasted some code from the email notification type. Seems to work quite well even though I don’t know Go...

Todo:

- [x] Documentation
- [x] Test if the configuration via environment variables works

Please provide me some feedback!

#264

### CODE REVIEW

**Comment by simonaronsson** [comment]
Awesome! Thank you for starting this work! I think `WATCHTOWER_NOTIFICATION_URL` and `--notification-url` would be preferable names for the params as all the other notifiers will be dropped over time.

**Comment by arnested** [comment]
Did you forget to git add new files? I can't seem to find the definition of `newShoutrrrNotifier()`?

Good work none the less!

**Comment by mbrandau** [comment]
> Awesome! Thank you for starting this work! I think `WATCHTOWER_NOTIFICATION_URL` and `--notification-url` would be preferable names for the params as all the other notifiers will be dropped over time.

I’ll change it.



> Did you forget to git add new files? I can't seem to find the definition of `newShoutrrrNotifier()`?
> 
> Good work none the less!

I did 😆

File: docs/notifications.md
Comment (arnested) [inline docs/notifications.md]: I think this will only result in one the URLs being set. One variable will override the other. I don't know how Viper does slices and environment variable (comma separated strings? space separated? ...?)

File: pkg/notifications/shoutrrr.go
Comment (arnested) [inline pkg/notifications/shoutrrr.go]: If you use the CreateSender approach you can get rid of the for loop and let Shoutrrr handle it for you:
```go
shoutrrr, _ := shoutrrr.CreateSender(e.Urls...);
err := shoutrrr.Send(msg, nil);
```

File: docs/notifications.md
Comment (mbrandau) [reply docs/notifications.md]: Confirmed. I will try out the slicing approach.

File: pkg/notifications/shoutrrr.go
Comment (mbrandau) [reply pkg/notifications/shoutrrr.go]: Done. I did not get rid of the loop though. I iterate over the errors now, so the user knows which service url failed.

File: docs/notifications.md
Comment (simonaronsson) [reply docs/notifications.md]: Hmm, I remember we did something around this in shoutrrr. I'll need to check it up though. From the top of my head, I think we decided on comma-separated URLs or something like that

Comment (mbrandau) [reply docs/notifications.md]: Space separation works. I’ll adjust the documentation.

File: pkg/notifications/shoutrrr.go
Comment (mbrandau) [reply pkg/notifications/shoutrrr.go]: Also I just moved `shoutrrr.CreateSender(e.Urls...)` to the `newShoutrrrNotifier` function so it gets reused.

Comment (arnested) [reply pkg/notifications/shoutrrr.go]: Ah, the Send() returns a slice of errors. Then looping over errors is fine.

**Comment by simonaronsson** [comment]
From my perspective, this actually looks quite nice. Great job! 👏🏼 Should I interpret your checklist as that you've been successful in sending watchtower notifications to slack through shoutrrr? If so, I'm impressed! 👍

**Comment by mbrandau** [comment]
Correct. So far I’ve tried Discord too but that didn't work yet.
Also I don’t think it makes a lot of sense to list all the services in this PR since that is all related to shoutrrr and not my implementation.

**Comment by simonaronsson** [comment]
> Correct. So far I’ve tried Discord too but that didn't work yet.
> Also I don’t think it makes a lot of sense to list all the services in this PR since that is all related to shoutrrr and not my implementation.

Agreed. I'm gonna take a look at your ticket in the shoutrrr repo. Let me get back on the discord support. :)

**Comment by simonaronsson** [comment]
Error in the discord service has now been resolved. 👍

**Comment by mbrandau** [comment]
I just updated the documentation to link to the overview page introduced by https://github.com/containrrr/shoutrrr/pull/18.
Can we merge this when https://github.com/containrrr/shoutrrr/pull/18 is ready or is there anything else this PR should cover?

**Comment by arnested** [comment]
Before merging I think the Shoutrrr dependency should be updated to latest version.

Most notably for these:

* fix referencing issue with the discord config: https://github.com/containrrr/shoutrrr/commit/6e7a09129f77d584620e2047562221a5cefdf684
* feat(services) add mattermost service: https://github.com/containrrr/shoutrrr/commit/f17c3e27d07ac823359a5153b06ed3c1bc7ee0bc
* feat(services): add hangouts chat service: https://github.com/containrrr/shoutrrr/commit/e46eebb058a87d2acd49d4b930b6664805fed4e8

**Comment by mbrandau** [comment]
@arnested can you check if I did it right?

**Comment by arnested** [comment]
Works like a charm, @mbrandau!

**Review by simonaronsson — APPROVED** [review]

**Comment by simonaronsson** [comment]
let's merge it and try it out for a while before releasing a new default version.
if you would like to help in trying it out, you can start watchtower from `containrrr/watchtower:latest-dev`

**Comment by mbrandau** [comment]
I will sure do that! 👍

**Comment by arnested** [comment]
Running `latest-dev` on two hosts now. One with Shoutrrr notifications to Slack and Hangouts. And one with Shoutrrr notifications to Gotify and Hangouts.

🚀

**Comment by simonaronsson** [comment]
I've been thinking that this will be the 1.0.0 release, once we know for sure it's stable 👍

**Comment by simonaronsson** [comment]
@all-contributors add @arnested for test, review

**Comment by simonaronsson** [comment]
@all-contributors add @mbrandau for test

**Pull request merged**
